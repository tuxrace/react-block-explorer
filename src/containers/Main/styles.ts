const styles = {
    container: {
        padding: 24,
        marginTop: 24
    },
    detailsCardStyle: {
        border: '1px solid lightgrey', borderRadius: 6, padding: 16
    },
    barChartStyle: {
        border: '1px solid lightgrey', borderRadius: 6, padding: 16, marginTop: 16
    }
}

export default styles
