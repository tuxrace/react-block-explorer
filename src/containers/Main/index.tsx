/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect, useState, useCallback } from 'react'
import Header from '../../components/Header'
import SearchBar from '../../components/SearchBar'
import DetailsCard from '../../components/DetailsCard'
import ToggleComponent from '../../components/ToggleComponent'
import BarChartComponent from '../../components/BarChartComponent'
import Container from '@material-ui/core/Container'
import Grid from '@material-ui/core/Grid'
import styles from './styles'
import { makeStyles, Typography } from '@material-ui/core'
import { getContracts } from '../../contractsService'
import { convertResults, convertStr } from '../../helpers'
import { tokenMap } from '../../config'
import { CoinTypes } from '../../types'
import DialogComponent from '../../components/DialogComponent'

const useStyle = makeStyles(styles)

const Main = () => {
    const [contract, setContract] = useState({} as any)
    const [loading, setLoading] = useState({} as any)
    const [open, setOpen] = useState(false)
    const [token, setToken] = useState<string>(tokenMap[CoinTypes.Alice])
    const classes = useStyle()
    
    const getData = useCallback(async () => {
        setLoading(true)
        const res = await getContracts({
            method: 'GetSmartContractInit',
            params: [token]
        })

        if (res?.error){
            setOpen(true)
            setLoading(false)
        }

        if (res?.result) {
            setLoading(false)
            setContract(convertResults(res?.result))
        }
    }, [token])

    useEffect(() => {
        getData()
    }, [token])
        
    const handleChange = (event: any, value: string) => {
        setToken(value)
    }

    const handleSearch = (value: string) => {
        setToken(convertStr(value))
    }

    const onModalClose = () => {
        setOpen(false)
    }

    return (
        <>
            <Header />
            <Container className={classes.container}>
                <Grid md={10} xs={12}>
                    <SearchBar handleSearch={handleSearch} />
                </Grid>
                <Grid md={12}>
                    <ToggleComponent selected={token} handleChange={handleChange}/>
                </Grid>
                <Grid container md={10} xs={12} className={classes.detailsCardStyle}>
                    <DetailsCard loading={loading} contract={contract} />
                </Grid>
                <Grid container md={10} xs={12} className={classes.barChartStyle}>
                    <BarChartComponent selected={token}/>
                </Grid>
                <DialogComponent 
                    open={open}
                    onClose={onModalClose}
                    title="Invalid Address"
                    content="The transaction address that you have entered on the address bar is somehow invalid. Please try again."
                />
            </Container>
        </>
    )
}

export default Main
