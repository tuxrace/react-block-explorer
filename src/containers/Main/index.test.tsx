import { render } from '@testing-library/react'
import Main from './index'

describe('Main component', () => {
    test('renders Main component', async () => {
        const {baseElement} = render(<Main />);
        expect(baseElement).toMatchSnapshot()
    })
})
