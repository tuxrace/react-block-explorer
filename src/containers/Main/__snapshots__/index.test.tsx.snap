// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component renders Main component 1`] = `
<body>
  <div>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-header-11 MuiPaper-elevation4"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6"
      >
        üõ°Ô∏è
         Clever Block Explorer
      </h6>
    </header>
    <div
      class="MuiContainer-root makeStyles-container-1 MuiContainer-maxWidthLg"
    >
      <div
        class="MuiGrid-root MuiGrid-grid-xs-12 MuiGrid-grid-md-10"
      >
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-searchStyle-4"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
            for="standard-start-adornment"
            id="standard-start-adornment-label"
          >
            transaction address
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              id="standard-start-adornment"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
              >
                üîé
              </p>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-7"
              >
                <span>
                  transaction address
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-grid-md-12"
      >
        <div
          class="MuiToggleButtonGroup-root makeStyles-toggleStyle-12"
          role="group"
        >
          <button
            aria-pressed="true"
            class="MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal Mui-selected"
            tabindex="0"
            type="button"
            value="5938fc8af82250ad6cf1da3bb92f4aa005cb2717"
          >
            <span
              class="MuiToggleButton-label"
            >
               
              AliceCoin
               
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-pressed="false"
            class="MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal"
            tabindex="0"
            type="button"
            value="acfbdcf4693b1780e474116211f7f4795f24e06a"
          >
            <span
              class="MuiToggleButton-label"
            >
               
              BobCoin
               
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-grid-xs-12 MuiGrid-grid-md-10"
        style="border: 1px solid lightgrey; border-radius: 6px; padding: 16px;"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
            role="progressbar"
            style="width: 24px; height: 24px;"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
